### ISSUE TITLE
Load DAGs per git-sync

### ISSUE THREAD

**Issue by adwk67** [stackabletech/airflow-operator#177]
Issue #150 introduces one possibility for general management of external resources: for DAGs, the [recommendation](https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html?highlight=git+sync#triggering-dags-after-changes) from Airflow is to use git-sync (see e.g. [here](https://blog.devgenius.io/setting-up-apache-airflow-on-kubernetes-with-gitsync-beaac2e397f3) for more info). This issue covers implementing git-sync in a sidecar to keep DAGs updated.

**Comment by fhennig** [comment]
My understanding is that #150 needs to be implemented together with this, to allow for multiple different DAG loading mechanisms, is that right? @adwk67 

Or is it possible to implement the git-sync mechanism without breaking changes?

**Comment by adwk67** [comment]
> My understanding is that #150 needs to be implemented together with this, to allow for multiple different DAG loading mechanisms, is that right? @adwk67
> Or is it possible to implement the git-sync mechanism without breaking changes?
> Another question about my understanding: The operator would get some git links, and clone the repos when starting up, presumably into an emptyDir. Is that right? No provisioning of volumes by the user required.

- This won't be a breaking change as we would be introducing a new CRD snippet, which should be applicable to other operators, too.
- this would be instead of, or alongside #150 i.e. they are independent of each other
- git-sync gets repo details and then periodically pulls changes to an emptyDir from where (in the case of airflow) DAGs can be mounted to the configured DAG folder

**Comment by lfrancke** [comment]
Are you two in agreement now? If so I'm happy to move it on

**Comment by sbernauer** [comment]
Just throwing in the idea of including all the stuff from `k8s.gcr.io/git-sync/git-sync:v3.2.2` in our Airflow Image. (We can e.g. extract it out of the upstream image)
We were able to do so in all other products AFAIK, eliminating the need to specify any image for the user. Another benefit is not overseeing the image when it comes to building arm images

**Comment by adwk67** [comment]
> Just throwing in the idea of including all the stuff from k8s.gcr.io/git-sync/git-sync:v3.2.2 in our Airflow Image
I like this idea and will aim to incorporate it.

**Comment by fhennig** [comment]
I agree

**Comment by backstreetkiwi** [comment]
Does that mean I can finally remove the node labels? (see unticked box in "acceptance criteria")? 
paging @adwk67

**Comment by sbernauer** [comment]
Just a side note: I we still rely on any nodeSelectors in any test for any operator we should probably use affinity instead - which is now supported :)

**Comment by adwk67** [comment]
> Does that mean I can finally remove the node labels? (see unticked box in "acceptance criteria")?

Yes

**Comment by adwk67** [comment]
> I just browsed through the docs and see that the "PersistentVolumeClaim" has gone missing. Has that feature been removed?

Yes, it was problematic and was one of the reasons we wanted to have git-sync instead.

**Comment by lfrancke** [comment]
In that case though that removal needs to be documented in the changelog

**Comment by adwk67** [comment]
The functionality is still there - we only removed the example/documentation. I've updated the changelog to reflect this in a separate PR.

**Comment by lfrancke** [comment]
Okay, I'm not sure I do understand why we remove the docs entirely then?
Do we want to remove the functionality in the future?

**Comment by adwk67** [comment]
Using PVCs for this kind of customer-facing thing is problematic as it depends on what type of PVC-access (read-write-many etc.) is available on a given cloud and if RWX is not available (as is the case with Ionos, for example) then node selection needs to be used to make sure this approach works. I don't think we should give the impression that we recommend or support it. And we didn't add any functionality for this: we just documented how it could be done.
